components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+                        onClick={() => {
+                            tryAuthorize(error?.data.host, error?.data.scopes).then((payload) =>
+                                setSelectAccountError(payload),
+                            );
+                        }}
+                    >
+                        Authorize with {error.data.host}
+                    </button>
+                </div>
+            );
+        case ErrorCodes.PERMISSION_DENIED:
+            return <p className="text-base text-gitpod-red w-96">Access is not allowed</p>;
+        case ErrorCodes.USER_BLOCKED:
+            window.location.href = "/blocked";
+            return <></>;
+        case ErrorCodes.NOT_FOUND:
+            return <p className="text-base text-gitpod-red w-96">{error.message}</p>;
+        case ErrorCodes.TOO_MANY_RUNNING_WORKSPACES:
+            return <LimitReachedParallelWorkspacesModal />;
+        case ErrorCodes.NOT_ENOUGH_CREDIT:
+            return <LimitReachedOutOfHours />;
+        case ErrorCodes.INVALID_COST_CENTER:
+            return (
+                <div>
+                    <p className="text-base text-gitpod-red w-96">
+                        The organization <b>{error.data}</b> is not valid.
+                    </p>
+                </div>
+            );
+        case ErrorCodes.PAYMENT_SPENDING_LIMIT_REACHED:
+            return <UsageLimitReachedModal hints={error?.data} />;
+        case ErrorCodes.PROJECT_REQUIRED:
+            return (
+                <p className="text-base text-gitpod-red w-96">
+                    <a className="gp-link" href="https://www.gitpod.io/docs/configure/projects">
+                        Learn more about projects
+                    </a>
+                </p>
+            );
+        default:
+            return <p className="text-base text-gitpod-red w-96">Unknown Error: {JSON.stringify(error, null, 2)}</p>;
+    }
+}
+
+function ExistingWorkspaceModal({
+    result,
+    onClose,
+    createWorkspace,
+}: {
+    result: WorkspaceCreationResult;
+    onClose: () => void;
+    createWorkspace: (opts: Omit<GitpodServer.CreateWorkspaceOptions, "contextUrl">) => void;
+}) {
+    return (
+        <Modal visible={true} closeable={false} onClose={onClose}>
+            <h3>Running Workspaces</h3>
+            <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-4 -mx-6 px-6 py-2">
+                <p className="mt-1 mb-2 text-base">
+                    You already have running workspaces with the same context. You can open an existing one or open a
+                    new workspace.
+                </p>
+                <>
+                    {result?.existingWorkspaces?.map((w) => {
+                        const normalizedContextUrl =
+                            ContextURL.getNormalizedURL(w.workspace)?.toString() || "undefined";
+                        return (
+                            <a
+                                href={
```

Comment:
We should add a `key` to the `<a>` here - any array of items rendered (usually in a `.map()` call) need a unique `key`. I'm not sure if there's a reliable unique property of `w` in this loop, but if so, that's a good value to use for `key`.